stdlib/experimental/prometheus/from.go

```diff
@@ -0,0 +1,339 @@
+	return FromPrometheusKind
+}
+
+func (s *FromPrometheusProcedureSpec) Copy() plan.ProcedureSpec {
+	ns := new(FromPrometheusProcedureSpec)
+	return ns
+}
+
+// uses a procedure spec to create a source object for flux runtime
+func createFromPrometheusSource(prSpec plan.ProcedureSpec, dsid execute.DatasetID, a execute.Administration) (execute.Source, error) {
+	spec, ok := prSpec.(*FromPrometheusProcedureSpec)
+	if !ok {
+		return nil, fmt.Errorf("invalid spec type %T", prSpec)
+	}
+
+	PrometheusIterator := PrometheusIterator{id: dsid, spec: spec, administration: a}
+
+	return execute.CreateSourceFromDecoder(&PrometheusIterator, dsid, a)
+}
+
+type PrometheusIterator struct {
+	id             execute.DatasetID
+	administration execute.Administration
+	spec           *FromPrometheusProcedureSpec
+	reader         *execute.RowReader
+
+	metrics []Metric
+	met     Metric
+	i       int
+	url     string
+	body    *http.Response
+}
+
+// prometheusScraper handles parsing prometheus metrics.
+// implements Scraper interfaces.
+// type prometheusScraper struct{}
+
+// ScraperTarget is a target to scrape
+type ScraperTarget struct {
+	Name string `json:"name"`
+	// Type ScraperType `json:"type"`
+	URL string `json:"url"`
+} // what do we need to scrape at bare minimum?
+
+// Metric is the default influx based metrics.
+type Metric struct {
+	Name      string                 `json:"name"`
+	Tags      map[string]string      `json:"tags"`
+	Fields    map[string]interface{} `json:"fields"`
+	Timestamp time.Time              `json:"timestamp"`
+	Type      string                 `json:"string"` // need to add other funcs for validation; first make sure this is right
+}
+
+func (p *PrometheusIterator) Connect(ctx context.Context) error {
+
+	return nil
+}
+
+func (p *PrometheusIterator) parse(r io.Reader, header http.Header) (err error) {
+	var parser expfmt.TextParser
+	now := time.Now()
+
+	mediatype, params, err := mime.ParseMediaType(header.Get("Content-Type"))
+	if err != nil {
+		return err
+	} // why do we need to parse the media type?
+
+	// Prepare output
+	metricFamilies := make(map[string]*dto.MetricFamily)
+	if mediatype == "application/vnd.google.protobuf" &&
+		// read about protobufs
+		params["encoding"] == "delimited" &&
+		params["proto"] == "io.prometheus.client.MetricFamily" {
+		for {
+			mf := &dto.MetricFamily{}
+			if _, err := pbutil.ReadDelimited(r, mf); err != nil {
+				if err == io.EOF {
+					break
+				}
+				return fmt.Errorf("reading metric family protocol buffer failed: %s", err)
+			}
+			metricFamilies[mf.GetName()] = mf
+		}
+	} else {
+		metricFamilies, err = parser.TextToMetricFamilies(r)
+		if err != nil {
+			return fmt.Errorf("reading text format failed: %s", err)
+		}
+	}
+
+	p.metrics = make([]Metric, 0)
+
+	// read metrics
+	for name, family := range metricFamilies {
+		for _, m := range family.Metric {
+			// read tags
+			tags := makeLabels(m)
+
+			// read fields
+			fields := make(map[string]interface{})
```

Comment:
this assignment is unnecessary.  if you want to create this variable and don't care about its initial value, use `var fields map[string]interface{}`